src/main.c

```diff
@@ -195,23 +201,89 @@ int _write(int file, char *ptr, int len)
 
 
 /**
+ * @brief Set left motor power.
+ *
+ * Power is set modulating the PWM signal sent to the motor driver.
+ *
+ * @param[in] power Power value from 0 to 1000.
+ */
+static void power_left(uint32_t power)
+{
+	timer_set_oc_value(TIM3, TIM_OC3, power);
+}
+
+
+/**
+ * @brief Set right motor power.
+ *
+ * Power is set modulating the PWM signal sent to the motor driver.
+ *
+ * @param[in] power Power value from 0 to 1000.
+ */
+static void power_right(uint32_t power)
+{
+	timer_set_oc_value(TIM3, TIM_OC4, power);
+}
+
+
+/**
+ * @brief Set driving direction to forward in both motors.
+ */
+static void drive_forward()
+{
+	gpio_clear(GPIOB, GPIO13 | GPIO15);
+	gpio_set(GPIOB, GPIO12 | GPIO14);
+}
+
+
+/**
+ * @brief Set driving direction to backward in both motors.
+ */
+static void drive_backward()
+{
+	gpio_clear(GPIOB, GPIO12 | GPIO14);
+	gpio_set(GPIOB, GPIO13 | GPIO15);
+}
+
+
+/**
+ * @brief Break both motors.
+ *
+ * Set driver controlling signals to high to short break the driver outputs.
+ * The break will be effective with any PWM input signal in the motor driver.
+ */
+static void drive_break()
+{
+	gpio_set(GPIOB, GPIO12 | GPIO13 | GPIO14 | GPIO15);
+}
+
+
+/**
  * @brief Initial setup and infinite wait.
  */
 int main(void)
 {
+	int j = 0;
+
 	setup_clock();
 	setup_gpio();
 	setup_usart();
 	setup_encoders();
 	setup_pwm();
 	setup_systick();
 
+	drive_forward();
+
 	while(1) {
 		int left = timer_get_counter(TIM1);
```

Comment:
Could you create a function to read the encoders to be more intuitive?

Reply:
Fixed with 3e276b010bed6f4f976419ac282958231c04595b.